src/Enum.php

```diff
@@ -2,425 +2,141 @@
-            throw new InvalidValueException($value, static::class);
-        }
-
-        if (is_null($index) || ! static::isValidIndex($index)) {
-            throw new InvalidIndexException($index, static::class);
-        }
-
-        $this->name = $name;
-        $this->value = $value;
-        $this->index = $index;
-    }
-
-    public function __call(string $name, array $arguments)
+    public static function toArray(): array
     {
-        if (static::startsWith($name, 'is')) {
-            return $this->isEqual(substr($name, 2));
-        }
+        $array = [];
 
-        if (static::isValidName($name)) {
-            return static::make($name);
+        foreach (static::resolveDefinition() as $definition) {
+            $array[$definition->value] = $definition->label;
         }
 
-        throw new BadMethodCallException('Call to undefined method '.static::class.'->'.$name.'()');
+        return $array;
     }
 
     /**
-     * @param string $name
-     * @param mixed[] $arguments
-     *
-     * @return \Spatie\Enum\Enumerable|bool
+     * @param string|int $value
      */
-    public static function __callStatic(string $name, array $arguments)
+    public function __construct($value)
     {
-        if (static::startsWith($name, 'is')) {
-            if (! isset($arguments[0])) {
-                throw new ArgumentCountError('Calling '.static::class.'::'.$name.'() in static context requires one argument');
-            }
+        $definition = $this->findDefinition($value);
 
-            return static::make($arguments[0])->$name();
-        }
+        if ($definition === null) {
+            $enumClass = static::class;
 
-        if (static::isValidName($name) || static::isValidValue($name)) {
-            return static::make($name);
+            throw new BadMethodCallException("There's no value {$value} defined for enum {$enumClass}, consider adding it in the docblock definition.");
         }
 
-        throw new BadMethodCallException('Call to undefined method '.static::class.'::'.$name.'()');
+        $this->value = $definition->value;
+        $this->label = $definition->label;
     }
 
-    public function __toString(): string
+    public function __get($name)
     {
-        return $this->getValue();
-    }
-
-    public function getIndex(): int
-    {
-        return $this->index;
-    }
-
-    public static function getIndices(): array
-    {
-        return array_column(static::resolve(), 'index');
-    }
+        if ($name === 'label') {
+            return $this->label;
+        }
 
-    public function getValue(): string
-    {
-        return $this->value;
+        if ($name === 'value') {
+            return $this->value;
+        }
     }
 
-    public static function getValues(): array
+    public static function __callStatic(string $name, array $arguments)
     {
-        return array_column(static::resolve(), 'value');
+        return new static($name);
```

Comment:
Right now you are also able to do `MyEnum::isDraft('draft')` - do we want to drop this possibility? This is useful if you want to test if any given value equals a fixed enum. For example to adjust validation rules in a request.

Reply:
I want this package to be focussed on the enum objects as much as possible, so that there's as little area as possible where you need to think about whether you're dealing with the object or the value — this was my main gripe with `myclabs/enum`, which is why I wanted to build this one. The more places support both the value- and object representation, the more place there are where you do need to worry about the distinction between them. 

Since there's a middleware in the laravel package which automatically transforms enums from requests to objects, I don't think we need to support the value representation. 

Looking forward to hearing your thoughts!

Reply:
Fien for me - a lot of my questions are primary to bring up the breaking points in this release and check if they are intentional or because of the current complexity simply forgotten.
I agree that the more we only use/allow an instance of the enum the better for us and the user regarding typing.